content/post/ml_talks_moscow.md

```diff
@@ -0,0 +1,163 @@
+---
+author: waren
+date: 2017-06-07
+title: "About our tech talks in Moscow"
+image: /post/ml_talks_moscow/back_audience.jpg
+description: "On June, 3-2017, we dedicated our regular [source{d} tech talks](http://talks.sourced.tech/) to Machine Learning and we chose to host the event in Moscow. For this conference, we invited speakers from Russia and abroad and gathered about 80 neural network aficionados in an former industrial area of the city. Let's make now a brief follow-up of this day."
+categories: ["technical"]
+---
+
+<style>
+p.dt {
+  margin-top: -16px;
+  font-style: italic;
+}
+.twitter-tweet {
+  margin-left: auto;
+  margin-right: auto;
+}
+.verbatim {
+  font-size: 0.85em;
+}
+</style>
+
+![picture of the audience during Egor's speech](/post/ml_talks_moscow/audience.jpg)
+
+On June, 3-2017, we dedicated our regular [source{d} tech talks](http://talks.sourced.tech/) to Machine Learning and we chose to host the event in Moscow. For this conference, we invited speakers from Russia and abroad and gathered about 80 neural network aficionados in an former industrial area of the city.
+
+# Day's programme
+
+To begin with, everybody joined together around a hearty welcome breakfast inside [KL10CH](https://www.linkedin.com/company/kl10ch) spaces in th city center.
+
+<img src="/post/ml_talks_moscow/breakfast.jpg" id="b2" alt="Photo during the welcome breakfast">
+<style>
+#b2 {
+  width: 600px;
+  height: 300px;
+  object-fit: contain;
+}
+</style>
+
+Then, after everybody woke up slowly, it was time for our CEO, [Eiso Kant](https://twitter.com/eisokant), to launch the main talks series. These lasted 45min each with time for Q&A. Furthermore, 2 lightning talks of 15min occurred between the main ones to address more specific topics.
+
+# Main talks
+
+### Statistical Analysis of Computer Program Text, _Charles Sutton_
+
+**_"Source code is a means of human communication"_**, with this first formula, the professor at University of Edinburgh, Charles Sutton, couldn't better start the day. He next laid out his statistical approach to analyze source code texts. In order to extract from scripts what he called, _implicit knowledge_, he introduced three innovative software engineering tools inspired from machine learning and natural language processing (NLP) techniques :
+
+* **Naturalize**, a probabilistic language model for source code which learns local coding conventions. It suggests renaming or reformatting changes so that your code would become more consistent.
+* **HAGGIS, Mining idioms for code** a system that learns local recurring syntactic patterns, which we call idioms, using a nonparametric Bayesian tree substitution grammar (TSG).
+* **Probabilistic API Miner (PAM)**, a near parameter-free probabilistic algorithm for mining the most interesting API patterns from a list of API call sequences. It resolves fundamental statistical pathologies like the formation of redundant or spurious sequences.
```

Comment:
AFAIK all 3 projects have web pages and github repos under http://mast-group.github.io/ so may be it's worth adding them here as a hypertext